File: src/forge/compiler/types/diagnostic.ts
export type DiagnosticSeverity = "error" | "warning" | "info";

export interface Diagnostic {
  severity: DiagnosticSeverity;
  code: string;
  message: string;
  file?: string;
  span?: { start: number; end: number };
  fixHint?: string;
  suggestedCommands?: string[];
  docs?: string[];
}

File: adapters/java/src/main/java/dev/forgeos/adapter/Diagnostic.java
package dev.forgeos.adapter;

import java.util.List;

public record Diagnostic(
    String severity,
    String code,
    String message,
    String file,
    String fixHint,
    List<String> docs
) {
  public static Diagnostic error(String code, String message) {
    return new Diagnostic("error", code, message, null, null, List.of("docs/forge-protocol.md"));
  }
}
